### PR TITLE
Improve some trivial things

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

### CODE REVIEW

**Review by insipx — APPROVED** [review]
TIL about `BoxFuture`

thanks !

File: substrate-archive/src/database/listener.rs [hunk 1]
```diff
@@ -41,21 +40,37 @@ pub struct Notif {
 	pub id: i32,
 }
 
+fn deserialize_number_from_string<'de, T, D>(deserializer: D) -> Result<T, D::Error>
```
Comment (dvdplm) [inline substrate-archive/src/database/listener.rs:43]: I had my eyes on this too, ty! :)

File: substrate-archive/src/lib.rs [hunk 2]
```diff
@@ -25,11 +25,11 @@ mod sql_block_builder;
 mod tasks;
 mod traits;
 
-pub use actors::System;
-pub use archive::Builder as ArchiveBuilder;
-pub use database::queries;
-pub use migrations::MigrationConfig;
-pub use traits::Archive;
+pub use self::actors::System;
```
Comment (dvdplm) [inline substrate-archive/src/lib.rs:28]: A matter of taste, but I prefer `pub use self::{ …, …, …, };`.

**Review by dvdplm — APPROVED** [review]
Thanks!

**Comment by koushiro** [comment]
@insipx @dvdplm PTAL and squash-merge it.

**Comment by koushiro** [comment]
@dvdplm Could you use the `squash-merge` for my PRs next time? I don't wanna push my trivial commits to the master branch

**Comment by dvdplm** [comment]
> @dvdplm Could you use the `squash-merge` for my PRs next time? I don't wanna push my trivial commits to the master branch

I was convinced that was the default, my bad.

**Pull request merged**